packages/client/README.md

```diff
@@ -114,257 +205,280 @@ const msg = {
+        },
+        to: {
+            timestamp: (Date.now() - 1000 * 60 * 5), // 5 minutes ago
+        },
+        // when using from and to the following parameters are optional
+        // but, if specified, both must be present
+        publisher: '0x12345...', 
+        msgChainId: 'ihuzetvg0c88ydd82z5o', 
     }
-})
+}, onMessage)
 ```
-
-To extract the session token from an authenticated client:
+If you choose one of the above resend options when subscribing, you can listen on the completion of this resend by doing the following:
 
 ```js
-const bearerToken = await client.session.getSessionToken()
+const sub = await streamr.subscribe(options)
+sub.onResent(() => {
+    console.log('All caught up and received all requested historical messages! Now switching to real time!')
+})
 ```
 
-Then for example,
+### Search Streams
+You can search for a stream which has the `term` string in it's id as follows:
 ```js
-    axios({
-        headers: {
-            Authorization: `Bearer ${bearerToken}`,
-        },
-        ...
-    )}
+const streams = await streamr.searchStreams('foo')
 ```
+You can query for the streams using an optional second parameter to fine-tune your search. A permission query searches over stream permissions. You can either query by direct permissions (which are explicitly granted to a user), or by all permissions (including public permissions, which apply to all users).
 
-Note, session tokens expire after four hours and may need to be refreshed.
-
-## Connecting
+To get all streams where a user has some direct permission. The `user` option can be omitted. In that case, it defaults to the authenticated user:
+```js 
+const streams = await streamr.searchStreams('foo', {
+    user: '0x12345...'
+})
+// Or, to query for all the streams accessible by the user
+const streams = await streamr.searchStreams('foo', {
+    user: '0x12345...',
+    allowPublic: true
+})
+```
 
-By default the client will automatically connect and disconnect as needed, ideally you should not need to manage connection state explicitly.
-Specifically, it will automatically connect when you publish or subscribe, and automatically disconnect once all subscriptions are removed and no
-messages were recently published. This behaviour can be disabled using the `autoConnect` & `autoDisconnect` options when creating a `new
-StreamrClient`. Explicit calls to either `connect()` or `disconnect()` will disable all `autoConnect` & `autoDisconnect` functionality, but they can
-be re-enabled by calling `enableAutoConnect()` or `enableAutoDisconnect()`.
+It is also possible to filter by specific permissions by using `allOf` and `anyOf` flags. Please prefer `allOf` to `anyOf` when possible as it has better query performance.
 
-Calls that need a connection, such as `publish` or `subscribe` will fail with an error if you are disconnected and autoConnect is disabled.
+If you want to find the streams you can exclusively subscribe to:
```

Comment:
For the both "you can publish to" examples we could leave out the search terms as it is more realistic use case and more in line what we describe in the sentence.

Reply:
What would that look like? Would `...` just do it, or does it become confusing?:
```js
const streams = await streamr.searchStreams(
    ...
    , {
      user: '0x12345...',
      allOf: [StreamPermission.SUBSCRIBE],
    }
 )
```

Reply:
I mean, use `undefined` as the search term. As we describe the use case as "streams you can subscribe to", not e.g. "streams you can subscribe to, and also containing term foobar":
https://github.com/streamr-dev/network-monorepo/blob/0c7e55c2b4417b3f049fc09241b28f9832ffef55/packages/client/src/StreamRegistry.ts#L259